docs/pages/reference/helm-reference/teleport-kube-agent.mdx

```diff
@@ -645,16 +645,20 @@ You can specify multiple selectors by adding elements to the list.
 |----------|---------------|
 | `string` | `nil`         |
 
-Normally the version of Teleport being used will match the version of the chart being installed. If you install chart version
-10.0.0, you'll be using Teleport 10.0.0.
+Normally the version of Teleport being used will match the version of the chart
+being installed. If you install chart version 10.0.0, you'll be using Teleport
+10.0.0. Upgrading the Helm chart will use the latest version from the repo.
 
-You can optionally override this to use a different published Teleport Docker image tag like `10.2.2` or `11`.
+You can optionally override this to use a different published Teleport Docker
+image tag like `10.1.2` or `11`.
 
-See [this link for information on Community Docker image versions](../../installation.mdx#docker).
+<Admonition type="danger">
+`teleportVersionOverride` MUST NOT be used to control the Teleport version.
```

Comment:
I think we should be clearer with this option.
If it's not meant to override the teleport version, than what should it be used for?

My guess is that we want to keep it because of dev builds, is that the only use case?

Reply:
This is also useful for development. There might be cases where you want to have the chart and teleport version mismatch in case of a bug. But basically the option is for users who know what they are doing. By default, no-one should use it if their need is "I want to run version X". 

We can't un-document because a lot of people are using it.

Reply:
I think I'm with Marco; If this is the case, I feel like we should just rename that parameter and remove all references to it from the docs. Leave it as a hidden field that can be set if needed, but I'm a huge proponent of removing all references to `teleportVersionOverride` and replacing them with guidance to use `helm --version` instead.

Reply:
Can we deprecate it? If the parameter disappears, people will have questions. I think it's better to leave a tombstone to guide users to the correct way of setting version

Reply:
I went with the suggestion of making this a development-only parameter, which kinda makes sense: https://github.com/gravitational/teleport/pull/25601#discussion_r1186436435